### ISSUE TITLE
WebVTT subtitles do not display in Firefox, but works fine with other browsers

### ISSUE THREAD

**Issue by anthony-evans** [ivyrze/omeka-s-module-transcript#6]
When displaying WebVTT subtitles using the latest version of Firefox, no subtitles are displayed, and there is just a rotating logo (see the right hand side of the screenshot below).

![image](https://github.com/ivyrze/omeka-s-module-transcript/assets/57145243/834aada8-06a4-4af3-8d1b-9fc39a455cd8)

However, the subtitles display perfectly fine in other browsers like Chrome or Safari.

Any suggestions would be most appreciated.

**Comment by ivyrze** [comment]
Thanks for writing in. I just pushed 18e50b1 which should fix this. If you're able to build from source and report back, let me know, otherwise I can bundle and publish it.

**Comment by anthony-evans** [comment]
Thankyou for getting back so quickly.  It seems to be a bit better now with the update, but we sometimes now get synchronisation issues and intermittent failure to load text in Firefox. But its an improvement nevertheless. 

Quite often waiting a few minutes and reloading the page fixes the text loading issue. We've noticed that on pages with multiple transcriptions, the first transcript file is loaded across all media players across the page (but only in Firefox) with some overlapping and out of sync issues after the first minute or two. 

We'll do some more tests over the weekend and will get back to you once we've found a pattern, as the problems are rather intermittent.

On Chrome and other browsers, the module works brilliantly though.

Thankyou for your time in looking into this. It is so much appreciated.

**Comment by ivyrze** [comment]
I'm considering doing a rewrite of the front-end in React, which I think will help work many of these issues, like the synchronization/data binding and having multiple player instances. I don't want to over-promise since bandwidth is limited, but I'm planning to research it more and pitch my team on it next week to see if we can make it happen.

**Comment by ivyrze** [comment]
I gave this a shot and things seem to be improved overall, but because there's been so many changes under the hood, I'm wondering if you could take another look and let me know what you think?

**Comment by anthony-evans** [comment]
> I gave this a shot and things seem to be improved overall, but because there's been so many changes under the hood, I'm wondering if you could take another look and let me know what you think?

Thankyou very much. It is working very well now and the Firefox problem seems to be resolved.

When the site is in developer mode, Omeka does give the following notice messages but the module is working perfectly otherwise without any errors when the site is in production mode...

```
Notice: Undefined index: quality in /var/www/html/omeka/modules/Transcript/view/common/media-embed.phtml on line 22

Notice: Undefined variable: poster in /var/www/html/omeka/modules/Transcript/view/common/media-embed.phtml on line 39
```

Thankyou for fixing the problem.

**Comment by ivyrze** [comment]
Okay great, thanks for testing it out for me. I'll get this patched up along with a few other things, and it'll get officially published as version 4.1 in a few weeks. Feel free to open up another issue if you come across anything else.
